src/components/tl/nccl/tl_nccl_coll.c

```diff
@@ -566,21 +606,17 @@ ucc_status_t ucc_tl_nccl_reduce_start(ucc_coll_task_t *coll_task)
 
 ucc_status_t ucc_tl_nccl_reduce_init(ucc_tl_nccl_task_t *task)
 {
-    ucc_tl_nccl_team_t *team = TASK_TEAM(task);
-    ucc_datatype_t dt;
+    ucc_coll_args_t *args = &TASK_ARGS(task);
 
-    dt = (TASK_ARGS(task).root == UCC_TL_TEAM_RANK(team))
-        ? TASK_ARGS(task).dst.info.datatype
-        : TASK_ARGS(task).src.info.datatype;
-
-    if (UCC_OK !=
-        ucc_nccl_check_dt_supported(dt, dt)) {
-        tl_debug(UCC_TASK_LIB(task), "dataype is not supported");
-        return UCC_ERR_NOT_SUPPORTED;
+    if (UCC_IS_ROOT(TASK_ARGS(task), UCC_TL_TEAM_RANK(TASK_TEAM(task)))) {
+        if (ucc_tl_nccl_check_and_convert_buffer_reduction(&args->dst.info,
+                                                           task) != UCC_OK) {
+            return UCC_ERR_NOT_SUPPORTED;
+        }
     }
 
-    if (ucc_to_nccl_reduce_op[TASK_ARGS(task).op] == ncclOpUnsupported) {
-        tl_debug(UCC_TASK_LIB(task), "reduction operation is not supported");
+    if (ucc_tl_nccl_check_and_convert_buffer_reduction(&args->src.info, task) !=
```

Comment:
if rank == root & IS_INPLaCE(args) -> will this be checking garbage value in args->src.info and potentially can incorrectly fallback?

Reply:
you're right. fixed